transforms/language/doc_quality/python/src/doc_quality_transform.py

```diff
@@ -0,0 +1,269 @@
+
+from argparse import ArgumentParser, Namespace
+from typing import Any
+
+import os
+import pyarrow as pa
+from data_processing.data_access import DataAccessFactory, DataAccess
+from data_processing.transform import AbstractTableTransform, TransformConfiguration
+from data_processing.utils import CLIArgumentProvider, TransformUtils, get_logger
+from doc_c4_statistics import (
+    c4_contain_pattern_ratio,
+    c4_contains_ldnoobw_words,
+    c4_load_ldnoobw_words,
+    c4_sentence_count,
+)
+from doc_Gopher_statistics import (
+    compute_average_japanese_sentence_length,
+    compute_bullet_point_ellipsis_alphabet_word_ratio,
+    compute_word_statistics,
+    contains_common_English_words,
+    find_first_japanese_alphabet_position,
+)
+from perplexity import (
+    get_model_file_name,
+    get_tokenizer_file_name,
+    KenLMModel,
+    MODEL_FILE_NAME_SUFFIX,
+    TOKENIZER_FILE_NAME_SUFFIX,
+)
+
+logger = get_logger(__name__)
+
+short_name = "docq"
+cli_prefix = f"{short_name}_"
+text_lang_key = "text_lang"
+doc_content_column_key = "doc_content_column"
+doc_id_column_key = "doc_id_column"
+bad_word_filepath_key = "bad_word_filepath"
+kenLM_model_key = "kenLM_model"
+text_lang_cli_param = f"{cli_prefix}{text_lang_key}"
+doc_content_column_cli_param = f"{cli_prefix}{doc_content_column_key}"
+doc_id_column_cli_param = f"{cli_prefix}{doc_id_column_key}"
+bad_word_filepath_cli_param = f"{cli_prefix}{bad_word_filepath_key}"
+kenLM_model_cli_param = f"{cli_prefix}{kenLM_model_key}"
+
+default_text_lang = "en"
+default_doc_content_column = "contents"
+default_doc_id_column = "documents_id"
+
+data_factory_internal_key = f"{cli_prefix}data_factory"
+files_to_use_internal_key = f"{cli_prefix}files_to_use"
+
+class DocQualityTransform(AbstractTableTransform):
+    """
+    Implements a simple copy of a pyarrow Table.
+    """
+
+    def __init__(self, config: dict[str, Any]):
+        """
+        Initialize based on the dictionary of configuration information.
+        This is generally called with configuration parsed from the CLI arguments defined
+        by the companion runtime, DocQualityTransformRuntime.
+        """
+        # Make sure that the param name corresponds to the name used in apply_input_params method
+        # of DocQualityTransformConfiguration class
+        super().__init__(config)
+        self.text_lang = config.get(text_lang_key, default_text_lang)
+        self.doc_content_column = config.get(doc_content_column_key, default_doc_content_column)
+        self.doc_id_column = config.get(doc_id_column_key, default_doc_id_column)
+        self.bad_word_filepath = config.get(bad_word_filepath_key, None)
+        if self.bad_word_filepath is None:
+            raise RuntimeError(f"Missing configuration value for key {bad_word_filepath_key}")
+
+        self.re_pattern = c4_load_ldnoobw_words(ft_lang=self.text_lang, file_path=self.bad_word_filepath)
```

Comment:
I think you need to use a DataAccessFactory so that this file could be loaded from S3 instead of just the local file system.  See loading resources doc https://github.com/IBM/data-prep-kit/blob/dev/data-processing-lib/doc/transform-external-resources.md

Reply:
Also, i think you should check to see if re_pattern is provided in the config by the caller (this could happen for multiple reasons).  If present, then the filepath is not required.  You also may find his useful when loading the file from this class's RayRuntime object which will allow loading and then setting this config value.

Reply:
done